src/libraries/System.Collections/src/System/Collections/Generic/ReadOnlySet.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Collections.ObjectModel
+{
+    /// <summary>Represents a read-only, generic set of values.</summary>
+    /// <typeparam name="T">The type of values in the set.</typeparam>
+    [DebuggerDisplay("Count = {Count}")]
+    public class ReadOnlySet<T> : IReadOnlySet<T>, ISet<T>, ICollection
```

Comment:
Should we consider it as `sealed`?

Reply:
`ReadOnlyCollection<T>` is unsealed as well.